### PR TITLE
build: Optimize is part of the single JAR

### LINKED ISSUE #22271: Include Optimize in the single JAR with the rest of the stack
### Context

Currently, Optimize is present within the monorepo but it's not inheriting from the zeebe parent POM (i.e., parent/pom.xml) and it is not being packaged as part of the single JAR with the rest of the stack. The objective of this task is to establish such parenting relationship for Optimize and ensure that Optimize is packages as part of the single application correctly.

Once this task is completed, it shall unblock the following tasks:
* https://github.com/camunda/camunda/issues/22294
* https://github.com/camunda/camunda/issues/22347
* https://github.com/camunda/camunda/issues/22351

### Acceptance Criteria

[ ] The `optimize` POM inherits from `parent/pom.xml`
[ ] Make sure that the Optimize distribution includes the dependencies versions defined in the Zeebe parent
[ ] `dist/` is the new distribution directory, instead of `optimize-distro`. Update: We keep `optimize-distro` for this release and will remove it with https://github.com/camunda/camunda/issues/25621
[ ] It is possible to launch Optimize with the rest of the stack
[ ] It is possible to launch Optimize as a standalone application
[ ] It's possible to launch the rest of the stack without Optimize
[ ] All of the TODOs in Optimize that were waiting for the completion of this task are resolved

### Links

* Part of https://github.com/camunda/product-hub/issues/2289

```[tasklist]
### Pull Requests
```

## Task Lifecycle

For managing the issue lifecycle, please use the workflow commands. You can see the available
commands by writing `/help` as a comment on this issue.

#### Engineering Review

- [ ] All code targeting the main branch has been reviewed by at least one Engineer
- [ ] The PR targeting the main branch has been approved by the reviewing Engineer
- [ ] If the API has changed, the API documentation is updated
- [ ] All other PRs (docs, controller etc.) in the breakdown have been approved

#### QA Review

- [ ] The change is implemented as described on all target environments/versions/modes
- [ ] The documentation changes are as expected

### Completion

- [ ] All Review stages are successfully completed
- [ ] All associated PRs are merged to the main branch(es) and stable branches
- [ ] The correct version labels are applied to the issue

### PR DESCRIPTION
**Pull request by buccarel**
## Description

With this PR, Optimize becomes part of the "single JAR" distribution of Camunda.

While still not part of the single application, being part of the single JAR means that:
* Optimize inherits from zeebe-parent
* the compliance rules defined for the whole project now also apply to Optimize
* the dependencies used by Optimize converge in versions with the dependencies used by the rest of the applications

To make this PR green and sound, some adjustments have been done:
[x] logback has been removed, in favor of log4j2, to resolve its dependency conflit
[x] the skipOptimize flag has been removed from the command lines that had it (it could be re-introduced in some CIs, if the build time exceeds the allowed threshold)
[x] a StandaloneOptimize class has been written

Some other adjustments still need to be done:
[x] Quartz is currently starting when the single application is launched, even though Optimize is not a part of it

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes https://github.com/camunda/camunda/issues/22271

### CODE REVIEW

File: optimize/backend/src/main/java/io/camunda/optimize/rest/cloud/CCSaaSNotificationClient.java
Comment (buccarel) [inline optimize/backend/src/main/java/io/camunda/optimize/rest/cloud/CCSaaSNotificationClient.java:84]: the code below was being detected as a bug, because the `synchronized` word was missing. It is my understanding that this was a bug for real, so I added the keyword in the places where it was detected as missing. From what I could see, nothing will change

File: .github/workflows/ci.yml
Comment (cmur2) [inline .github/workflows/ci.yml:12]: 🔧 Please avoid applying auto-formatting to files like this (I assume via an editor that formats on save according to some internal criteria) which just introduces noise in the Git history by adding spaces and makes review harder.

File: optimize/util/optimize-commons/src/main/java/io/camunda/optimize/util/tomcat/LoggingConfigurationReader.java
Comment (romansmirnov) [reply optimize/util/optimize-commons/src/main/java/io/camunda/optimize/util/tomcat/LoggingConfigurationReader.java:28]: Also, when starting via `optimize-startup.sh` it logs

```
Starting Camunda Optimize 8.7.0-SNAPSHOT...

SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.
Standard Commons Logging discovery in action with spring-jcl: please remove commons-logging.jar from classpath in order to avoid potential conflicts
2024-12-09 13:32:49.006  INFO 93185 --- [kground-preinit] o.h.v.i.u.Version                        : HV000001: Hibernate Validator 8.0.1.Final
2024-12-09 13:32:49.029  INFO 93185 --- [           main] i.c.o.Main                               : Starting Main v8.7.0-SNAPSHOT using Java 21.0.4 with PID 93185 (/Users/roman.smirnov/Development/projects/camunda/optimize-distro/target/camunda-optimize-8.7.0-SNAPSHOT-production/optimize-backend-8.7.0-SNAPSHOT.jar started by roman.smirnov in /Users/roman.smirnov/Development/projects/camunda/optimize-distro/target/camunda-optimize-8.7.0-SNAPSHOT-production)
2024-12-09 13:32:49.030  INFO 93185 --- [           main] i.c.o.Main                               : The following 1 profile is active: "ccsm"
2024-12-09 13:32:49.809  INFO 93185 --- [           main] o.s.b.w.e.t.TomcatWebServer              : Tomcat initialized with ports 8090 (http), 8091 (https)
Dez. 09, 2024 1:32:49 PM org.apache.coyote.AbstractProtocol init
INFORMATION: Initializing ProtocolHandler ["http-nio-8090"]
Dez. 09, 2024 1:32:49 PM org.apache.coyote.AbstractProtocol init
```

File: optimize/backend/pom.xml
Comment (romansmirnov) [inline optimize/backend/pom.xml:674]: ❓ The Single Jar/App Stack uses `commons-collections4`. Could the usage of `commons-collections` be removed and replaced?

**Review by romansmirnov — COMMENTED** [review]
@buccarel, thanks for the PR. I went through the changes and added a few comments. Let's sync and go through them to decide which are necessary right away and which could be done later.

A general question would be: This PR introduces the `StandaloneOptimize` class, but it is not used? I would suggest using it asap to avoid having to different main classes.

File: optimize/util/optimize-commons/src/main/java/io/camunda/optimize/util/tomcat/LoggingConfigurationReader.java
Comment (buccarel) [reply optimize/util/optimize-commons/src/main/java/io/camunda/optimize/util/tomcat/LoggingConfigurationReader.java:28]: About that `environment-log4j2.xml`, that is a list of file names that the customer can define and they are loaded in order. For the moment I simply translated into log4j2 the old behavior optimize had with the old logging system

File: optimize/util/optimize-commons/src/test/resources/logback-test.xml
Comment (buccarel) [reply optimize/util/optimize-commons/src/test/resources/logback-test.xml:1]: Yes!
Will delete, thanks!

File: optimize/backend/src/main/java/io/camunda/optimize/rest/cloud/CCSaaSNotificationClient.java
Comment (buccarel) [reply optimize/backend/src/main/java/io/camunda/optimize/rest/cloud/CCSaaSNotificationClient.java:84]: I thought the critical section was the access to the `accessToken`, so I thought that the bug was legit.
I'll check again and revert this, in case.

I am assuming it is possible to override or ignore the error I'm getting?

File: optimize/backend/pom.xml
Comment (buccarel) [reply optimize/backend/pom.xml]: I could find one single use of commons-collections. I replaced it using commons-collections4. This should allow me to remove the old library.

File: optimize/pom.xml
Comment (buccarel) [reply optimize/pom.xml:56]: Done: I replaced the versions explicitly declared with one of the following options:
* I removed the version tag completely
* I used the version tag used in parent/pom.xml

Only when I couldn't find a version already declared, I kept the value of optimize

**Review by romansmirnov — APPROVED** [review]
LGTM. I have tested it a bit locally, and code-wise it looks fine. Only one small finding: When starting the broker (as Single Application with the profile, or standalone) and going to `localhost:8080`, it returns Optimize's error page:

> You have reached an invalid page in Optimize. Please use a different URL

This is not blocking this PR, would be great if you could handle this with another issue.

Besides that, I went through the failing CI which is a bit concerning. Could you please re-run them?

**Comment by romansmirnov** [comment]
Besides that, please consider also @cmur2 comment.

File: .github/workflows/ci.yml
Comment (buccarel) [reply .github/workflows/ci.yml]: I undid a good number of auto-save changes.
I decided to keep the bigger ones against the POMs, because those optimize files are now actually organized better than before.

Sorry about it. I deactivated the auto-format option in intelliJ now

File: dist/src/main/java/io/camunda/application/StandaloneOptimize.java
Comment (buccarel) [reply dist/src/main/java/io/camunda/application/StandaloneOptimize.java:25]: As discussed in slack, for the moment we're keeping it, but we're removing it after the merge of this PR as a follow up task

**Pull request merged**
